src/tonic/api/raft_api.rs

```diff
@@ -1,29 +1,36 @@
 use std::sync::mpsc::SyncSender;
-use std::sync::{Arc, Mutex};
+use std::sync::Mutex;
 
 use api::grpc::qdrant::raft_server::Raft;
 use api::grpc::qdrant::{
     AddPeerToKnownMessage, AllPeers, Peer, PeerId, RaftMessage as RaftMessageBytes, Uri as UriStr,
 };
+use collection::shard::ChannelService;
 use itertools::Itertools;
 use raft::eraftpb::{ConfChangeType, ConfChangeV2, Message as RaftMessage};
 use storage::content_manager::consensus_ops::ConsensusOperations;
-use storage::dispatcher::Dispatcher;
+use storage::content_manager::consensus_state::ConsensusStateRef;
 use tonic::transport::Uri;
 use tonic::{async_trait, Request, Response, Status};
 
 use crate::consensus;
 
 pub struct RaftService {
     message_sender: Mutex<SyncSender<consensus::Message>>,
-    dispatcher: Arc<Dispatcher>,
+    channel_service: ChannelService,
+    consensus_state: ConsensusStateRef,
 }
 
 impl RaftService {
-    pub fn new(sender: SyncSender<consensus::Message>, dispatcher: Arc<Dispatcher>) -> Self {
+    pub fn new(
+        sender: SyncSender<consensus::Message>,
+        channel_service: ChannelService,
+        consensus_state: ConsensusStateRef,
```

Comment:
```suggestion
        consensus_state: ConsensusStateRef,
```
There is a `peer_address_by_id` method on `ConsensusStateRef`